pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -482,47 +501,32 @@ def _monitor_candidates(self, trainer):
         monitor_candidates.update(step=trainer.global_step, epoch=trainer.current_epoch)
         return monitor_candidates
 
-    def _save_last_checkpoint(self, trainer, ckpt_name_metrics):
-        should_save_last = self.monitor is None or self.save_last
-        if not should_save_last:
+    def _save_last_checkpoint(self, trainer, monitor_candidates: Dict[str, Any]):
+        if not self.save_last:
```

Comment:
so this is the bugfix here? you are removing the condition on `monitor is None` and moving it to the `_save_none_monitor_checkpoint`?

Reply:
Yes, separating the logic into two functions